packages/contracts/test/attestation.test.ts

```diff
@@ -138,33 +119,114 @@ describe('Attestations', function () {
         expect(tx)
             .to.emit(unirepContract, 'AttestationSubmitted')
             .withArgs(
-                1,
+                epoch,
                 epochKey,
-                accounts[1].address,
+                attester.address,
                 posRep,
                 negRep,
                 graffiti,
                 timestamp
             )
+        {
+            const tx = await unirepContract.buildHashchain(
```

Comment:
Why are we doing the hashchain logic here? We've already tested what we're looking for

Reply:
`buildHashchain` is not checked in `@unirep/contracts`
I think `@unirep/contracts` can cover all the functions in `Unirep.sol`

Reply:
Ok but I think that should be in a different test case

Reply:
Create the issue here https://github.com/Unirep/Unirep/issues/163
We can do that in another PR

Reply:
Ok, in the meantime we should take out this extra test logic.